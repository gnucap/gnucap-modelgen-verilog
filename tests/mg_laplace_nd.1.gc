attach ./modelgen_0.so
attach ../vams/vsine.so

verilog

`modelgen
module test_lap_nd1(p, n);
	(* desc="" *) real dv;
	electrical p, n;
	analog begin
		dv = laplace_nd(V(p) - V(n), '{1}, '{1});
	end
endmodule

!make test_lap_nd1.so > /dev/null
attach ./test_lap_nd1.so

verilog
parameter r=1
parameter v=0

test_lap_nd1 #() dut(1,0);
vsine #(.dc(v), .ampl(1)) v1(1, 0);

list

print tran v(1) dv(dut) i(dut._b*) vin(dut.*) iter(0)
// y(dut.ddt_0) iof(dut.ddt_0) iter(0)
tran 1 .1 trace=n
status notime
end
