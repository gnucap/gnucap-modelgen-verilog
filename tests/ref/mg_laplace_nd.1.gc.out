parameter r=1 
parameter v=0 
test_lap_nd1 #() dut (.p(1),.n(0));
vsine #(.ampl(1),.dc(v)) v1 (.p(1),.n(0));
#Time       v(1)       dv1(dut)   dv2(dut)   dv3(dut)   dv4(dut)   dv5(dut)  
 0.         0.         0.         0.         0.         0.         0.        
 0.1        0.58779    0.014905   0.02981    0.044715   0.05962    0.074525  
 0.2        0.95106    0.052895   0.10579    0.15869    0.21158    0.26448   
 0.3        0.95106    0.097895   0.19579    0.29368    0.39158    0.48947   
 0.4        0.58779    0.13156    0.26311    0.39467    0.52623    0.65779   
 0.5       -7.f        0.13975    0.27951    0.41926    0.55902    0.69877   
 0.6       -0.58779    0.1186     0.2372     0.3558     0.4744     0.593     
 0.7       -0.95106    0.075283   0.15057    0.22585    0.30113    0.37641   
 0.8       -0.95106    0.025217   0.050434   0.075652   0.10087    0.12609   
 0.9       -0.58779   -0.01445   -0.0289    -0.04335   -0.0578    -0.07225   
 1.         7.f       -0.028412  -0.056824  -0.085237  -0.11365   -0.14206   
Gnucap   System status
iterations: op=0, dc=0, tran=134, fourier=0, total=168
transient timesteps: accepted=30, rejected=4, total=34
nodes: user=1, subckt=0, model=11, total=12
dctran density=30.6%, ac density=23.6%
